src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -167,6 +168,27 @@ export class ChatWidget extends Disposable implements IChatWidget {
 		this.inputPart.focus();
 	}
 
+	focusWithId(id: string, type: 'next' | 'previous'): void {
+		const items = this.viewModel?.getItems();
+		if (!items) {
+			return;
+		}
+		const focusedElement = items.find(i => i.id === id);
+		if (!focusedElement) {
+			return;
+		}
+		const responseItems = items.filter(i => isResponseVM(i));
+		const currentlyFocusedIndex = responseItems?.indexOf(focusedElement);
```

Comment:
And this is the index of the item you want to focus right?

Reply:
And I don't see how `responseItems` would be undefined

Reply:
no, this is the index of the currently focused element.